backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/RemoveMdevCommand.java

```diff
@@ -0,0 +1,39 @@
+package org.ovirt.engine.core.bll;
+
+import javax.inject.Inject;
+
+import org.ovirt.engine.core.bll.context.CommandContext;
+import org.ovirt.engine.core.bll.utils.CompensationUtils;
+import org.ovirt.engine.core.common.action.MdevParameters;
+import org.ovirt.engine.core.common.businessentities.VmDevice;
+import org.ovirt.engine.core.common.businessentities.VmDeviceId;
+import org.ovirt.engine.core.common.errors.EngineMessage;
+import org.ovirt.engine.core.dao.VmDeviceDao;
+
+public class RemoveMdevCommand extends AbstractMdevCommand<MdevParameters> {
+
+    @Inject
+    private VmDeviceDao vmDeviceDao;
+
+    public RemoveMdevCommand(MdevParameters parameters, CommandContext cmdContext) {
+        super(parameters, cmdContext);
+    }
+
+    @Override
+    protected void executeCommand() {
+        VmDeviceId deviceId = getParameters().getDevice().getId();
+        CompensationUtils.removeEntity(deviceId, vmDeviceDao, getCompensationContextIfEnabledByCaller());
+        compensationStateChanged();
```

Comment:
this is a transactional command so no need to use compensation here

Reply:
I see.  If it is mostly harmless to use the compensation, would it be OK to fix it in a separate PR? I'd rather avoid another verification round at this moment.

Reply:
yes, sure

Reply:
Handled, together with other changes, in https://github.com/oVirt/ovirt-engine/pull/166